core/templates/pages/splash-page/splash-page.component.ts

```diff
@@ -29,47 +30,74 @@ const splashConstants = require('constants.ts');
   template: require('./splash-page.component.html'),
   controller: [
-    '$rootScope', '$timeout', 'LoaderService', 'SiteAnalyticsService',
-    'UrlInterpolationService', 'UserService', 'WindowRef',
+    '$rootScope', 'LoaderService', 'SiteAnalyticsService',
+    'UrlInterpolationService', 'UserService', 'WindowDimensionsService',
     function(
-        $rootScope, $timeout, LoaderService, SiteAnalyticsService,
-        UrlInterpolationService, UserService, WindowRef) {
+        $rootScope, LoaderService, SiteAnalyticsService,
+        UrlInterpolationService, UserService, WindowDimensionsService) {
       var ctrl = this;
       ctrl.getStaticImageUrl = function(imagePath) {
         return UrlInterpolationService.getStaticImageUrl(imagePath);
       };
-      ctrl.getStaticSubjectImageUrl = function(subjectName) {
-        return UrlInterpolationService.getStaticImageUrl(
-          '/subjects/' + subjectName + '.svg');
-      };
-
-      ctrl.onRedirectToLogin = function(destinationUrl) {
-        SiteAnalyticsService.registerStartLoginEvent(
-          'splashPageCreateExplorationButton');
-        $timeout(function() {
-          WindowRef.nativeWindow.location = destinationUrl;
-        }, 150);
-        return false;
-      };
 
       ctrl.onClickBrowseLessonsButton = function() {
         SiteAnalyticsService.registerClickBrowseLessonsButtonEvent();
-        $timeout(function() {
-          WindowRef.nativeWindow.location = (
-            `/learn/${splashConstants.DEFAULT_CLASSROOM_URL_FRAGMENT}`);
-        }, 150);
         return false;
       };
 
-      ctrl.onClickCreateExplorationButton = function() {
-        SiteAnalyticsService.registerClickCreateExplorationButtonEvent();
-        $timeout(function() {
-          WindowRef.nativeWindow.location = '/creator-dashboard?mode=create';
-        }, 150);
-        return false;
+      ctrl.getMainBackgroundUrl = function() {
+        if (ctrl.isWindowNarrow()) {
+          return ctrl.getStaticImageUrl('/splash/splashMainMobile.png');
+        } else {
+          return ctrl.getStaticImageUrl('/splash/splashMainDesktop.png');
+        }
+      };
+
+      ctrl.isWindowNarrow = function() {
+        return WindowDimensionsService.isWindowNarrow();
+      };
+
+      // The 2 functions below are to cycle between values: 0, 1, 2 for
+      // testimonialId.
+      ctrl.incrementTestimonialId = function() {
+        // This makes sure that incrementing from 2, returns 0 instead of 3,
+        // since we want the testimonials to cycle through.
+        ctrl.testimonialId = (ctrl.testimonialId + 1) % 3;
+      };
+
+      ctrl.decrementTestimonialId = function() {
+        // This makes sure that decrementing from 0, returns 2 instead of -1,
+        // since we want the testimonials to cycle through.
+        ctrl.testimonialId = (((ctrl.testimonialId - 1) % 3) + 3) % 3;
+      };
+
+      ctrl.getTestimonialBackgroundUrl = function() {
+        if (ctrl.isWindowNarrow()) {
```

Comment:
Couldn't you just use CSS media queries to achieve this? Ditto below.

Reply:
You mean, move all of this to CSS? I couldn't do that since the URL itself requires calling `getStaticImageUrl()` function, which I don't think is possible via CSS.

Reply:
But is it possible to hardcode the URL in the CSS?

Reply:
It would be possible to hardcode, but we don't do that for images in general, so not sure if we should do this here.

Reply:
Per our discussion, we do it already (e.g. for the svg diagram editor and music notes input), so I think doing it here would make sense.